app/code/Magento/CatalogSearch/Model/ResourceModel/Fulltext/Collection.php

```diff
@@ -265,7 +265,7 @@ public function addFieldToFilter($field, $condition = null)
 
         $this->getSearchCriteriaBuilder();
         $this->getFilterBuilder();
-        if (!is_array($condition) || !in_array(key($condition), ['from', 'to'])) {
+        if (!is_array($condition) || !in_array((string)key($condition), ['from', 'to'])) {
```

Comment:
Okay it seems that the build fails with pretty cryptic error. Considerign that develop branch itself is all in red, I would rather wait the PR target to go to be back in green before looking into that one. 

Considering the narrow scope of the change I really doubt that it's caused by this branch.